File: leetcode/2593.py
class Solution:
    def findScore(self, nums: List[int]) -> int:
        sorted_nums = sorted([(num, i) for i, num in enumerate(nums)])
        score = 0
        for (num, i) in sorted_nums:
            if nums[i]:
                score += num
                if i+1 < len(nums):
                    nums[i+1] = 0
                if i-1 >= 0:
                    nums[i-1] = 0
        return score

File: leetcode/2593.cpp
class Solution
{
public:
    long long findScore(vector<int> &nums)
    {
        set<pair<int, int>> s;
        for (int i = 0; i < nums.size(); ++i)
            s.insert({nums[i], i});
        long long ans = 0;
        while (s.size())
        {
            auto cur = *s.begin();
            s.erase(cur);
            ans += cur.first;
            for (auto index : {cur.second - 1, cur.second + 1})
                if (index >= 0 && index < nums.size())
                    s.erase({nums[index], index});
        }
        return ans;
    }
};